File: feature-auth/src/main/java/com/levnovikov/feature_auth/AuthView.java
package com.levnovikov.feature_auth;

import android.content.Context;
import android.support.constraint.ConstraintLayout;
import android.util.AttributeSet;

import com.levnovikov.feature_auth.view_model.AuthActions;
import com.levnovikov.feature_auth.view_model.AuthViewModel;

import javax.inject.Inject;

/**
 * Created by lev.novikov
 * Date: 29/1/18.
 */

public class AuthView extends ConstraintLayout {
    public AuthView(Context context) {
        super(context);
    }

    public AuthView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public AuthView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Inject
    AuthActions vmActions;

    @Inject
    AuthViewModel vm;

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        vmActions.onGetActive();
    }
}

File: feature-auth/src/main/java/com/levnovikov/feature_auth/AuthView.kt
package com.levnovikov.feature_auth

import android.content.Context
import android.support.constraint.ConstraintLayout
import android.util.AttributeSet

import com.levnovikov.feature_auth.view_model.AuthActions
import com.levnovikov.feature_auth.view_model.AuthViewModel

import javax.inject.Inject

/**
 * Created by lev.novikov
 * Date: 29/1/18.
 */

class AuthView : ConstraintLayout {

    @Inject
    lateinit var vmActions: AuthActions

    @Inject
    lateinit var vm: AuthViewModel

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        vmActions.onGetActive()
    }
}
